c++/src/kj/compat/http.h

```diff
@@ -674,12 +674,158 @@ struct HttpConnectSettings {
   // * the underlying HttpClient does not support the startTls mechanism
   // * `useTls` has been set to `true` and so TLS has already been started
   //
-  // The callback function itself can be used to initiate a TLS handshake on the
-  // connection at any arbitrary point. The function returns an AsyncIoStream that is a secure
-  // TLS stream. This mechanism is required for certain protocols, more info can be found on
+  // The callback function itself can be called to initiate a TLS handshake on the connection in
+  // between write() operations. It is not allowed to initiate a TLS handshake while a write
+  // operation or a pump operation to the connection exists. Read operations are not subject to
+  // the same constraint, however: implementations are required to be able to handle TLS
+  // initiation while a read operation or pump operation from the connection exists. Once the
+  // promise returned from the callback is fulfilled, the connection has become a secure stream,
+  // and write operations are once again permitted. The StringPtr parameter to the callback,
+  // expectedServerHostname (TODO: must remain valid for the lifetime of the promise? may be
+  // dropped after the function synchronously returns?).
+
+
+  // This mechanism is required for certain protocols, more info can be found on
   // https://en.wikipedia.org/wiki/Opportunistic_TLS.
 };
 
+class PausableAsyncIoStream final: public kj::AsyncIoStream {
```

Comment:
Needs a doc comment; probably should be mentioned in the `tlsStarter` doc comment, too.